### ISSUE TITLE
Better source categorization

### ISSUE THREAD

**Issue by bhousel** [osmlab/editor-layer-index#136]
per @nicolas17 #134 

> I think imagery layers (like satellite) should be here, external raster layers (like some government's open data) should be here, and OSM-based layers (like QA) should be here. But there should be a field in the JSON saying in what "category" the layer is, to allow filtering if wanted.

**Comment by grischard** [comment]
Why and when would this kind of filtering be wanted?

**Comment by simonpoole** [comment]
For example a website wanting to show aerial imagery but not maps ....

**Comment by Klumbumbus** [comment]
see also https://josm.openstreetmap.de/ticket/16103

**Comment by Marc-marc-marc** [comment]
for the key name:
@grischard proposes `category`in #266 that look like fine.
for the value : 
@grischard proposes `osm`in #266 for render made from osm data. it would have the advantage of involving an (c) openstreetmap contributors.
@stoecker [proposes](https://josm.openstreetmap.de/ticket/16103) `map` and `photo`.
it look like a good idea to also add those 2 values (map for not-osm-based map like official one, and photo for sat and drone imagery).

did we need a value for derivative works like QA ?

**Comment by Klumbumbus** [comment]
>@stoecker proposes map and photo.
The disuccion in this ticket went a bit further

**Comment by Marc-marc-marc** [comment]
@Klumbumbus you're right. I forget to check it again before posting.
from josm ticket, i like map, osmbasedmap, photo, other
I dislike historicmap. should it be enough to use "date" for historic category ? I think so.
question remain for QA : is it a good idea to put them into osmbasedmap ? a layer which displays for example the error points found by Osmose, it is not really a map, but still osm-based.
I hope that the discussions on the two projects will continue to be mutually enriching in order to have a common vision that will facilitate everyone's life.

**Comment by Marc-marc-marc** [comment]
so what ? what did you ting if : 
- we add a key category with value map, osmbasedmap (or osmbased ?), photo, other ? qa ?
- we add not-the-lasted=no to allow app to hide or group previous versions of a layer ?

**Comment by Klumbumbus** [comment]
**`category`** was added just today to josm https://josm.openstreetmap.de/ticket/16103#comment:28 with the following supported values:
**`photo`** aerial or satellite photo, **`map`** a map, **`historicmap`** historic or otherwise outdated map, **`osmbasedmap`** map based on OSM data, **`historicphoto`** historic or otherwise outdated aerial or satellite photo, **`other`** any other type

**Comment by Marc-marc-marc** [comment]
yes but as I said on josm ticket, mixing category and not-the-lasted is not a good idea
and in addition to that, historic in not a good key for "not the last update available"
it's why I propose another word but that still allow a eli<>josm (easy) match
the question remain for layer like osmose : it's osmbase but not a map. should it be category=osmbase ? other ? qa ?

**Comment by Marc-marc-marc** [comment]
yes you can of course use only start_date/end_date to find the lasted layer but it's hard to find witch layers cover the same feature or not.
for ex : 
- layer id foo 2016 and layer id bar 2017 are the same feature for the end user or not ?
for a human, it's easy to have an idea, for an app, it's not easy (you need to compare shapes for ex)
- layers id municipaly1 date 2016 layer id municipality2 date 2016 layer id municipality1+2 date 2017
if you want to use only the lasted, it's not easy for a query to find that the municipaly1and municipality2 layers can be dropped due the fact the last one cover the same shape+feature.
- how to find that another layer show not hidden because the feature are not the same (infrared <> true color)
that need a lot of keys, a lot of work to fill all key....or put the result in `not-the-lasted` when an old layer have the same feature, nearly the same shape than a new one

**Comment by bhousel** [comment]
@Marc-marc-marc I'm confused what you mean by "which layers cover the same feature".. The three bullet points all relate to comparing geometry, and this doesn't seem like a very challenging issue.  In fact using the imagery index in the first place requires an application to do spatial queries around the place of interest to know which layers apply there. 

My plan for dealing with this in iD is to just throw the whole index into [which-polygon](https://github.com/mapbox/which-polygon) (which is itself based on an [rbush](https://github.com/mourner/rbush) index) and ask it which layers cover the area where the user is looking.

**Comment by Marc-marc-marc** [comment]
in this case you don't need `not-the-lasted` let's move forward without

and for `category` value ? do we need to make a separate category for the different osm-based rendering layers (for ex `osmbasedmap`) <> qa tools (for ex `qa`?) or we put all into a `osmbased`
imho it's better to split render (an "display only" app can get those) and qa layer (usefull as overlay when editing the map)

**Comment by simonpoole** [comment]
I would suggest using the category values from JOSM plus a value for qa layers (regardless of source), could we get buy in from the JOSM devs @don-vip for a common value for that?

**Comment by grischard** [comment]
Replicating josm's values sounds good to me. I'm ok with having a 'qa' category if JOSM devs agree to have it too.

So as far as I can tell, we need to:

- [x] find consensus on 'qa' with team JOSM
- [x] update schema
- [x] update conversion scripts on the ELI side
- [x] ~update geojson conversion script on the JOSM side?~

**Comment by don-vip** [comment]
I'm ok with 'qa' :)

**Comment by Klumbumbus** [comment]
Recent discussions in JOSMs trac in https://josm.openstreetmap.de/ticket/18172

**Comment by grischard** [comment]
I'm cool with 'terrain' too if JOSM decides to have it.

**Comment by simon04** [comment]
FYI, JOSM has added `elevation` and `qa` in https://josm.openstreetmap.de/changeset/15658/josm

**Comment by grischard** [comment]
All right, now let's get #733 ready :)

**Comment by grischard** [comment]
So we now need to update the conversion scripts but also @rbuffat's strict check script.

@Klumbumbus will the JOSM ImageryCompare compare the categories?

**Comment by don-vip** [comment]
> @Klumbumbus will the JOSM ImageryCompare compare the categories?

Yes: https://josm.openstreetmap.de/changeset/15692/josm/

**Comment by grischard** [comment]
This is now a long ticket, so to summarise what's left to be done:

- [x] update ELI conversion scripts
- [ ] update the strict check.py

**Comment by simonpoole** [comment]
> 
> 
> This is now a long ticket, so to summarise what's left to be done:
> 
>     * [ ]  update ELI conversion scripts
Didn't I already do that (at least the PR was merged :-))? It doesn't make sense to update the legacy format one, because ... legacy format that doesn't include the category (and I suspect nobody actually uses it any more).
> 
>     * [ ]  update the strict check.py

**Comment by grischard** [comment]
~convert_individual.py and convert_xml.py would need to understand the 'category' attribute too.~

Edit: just the strict check.py then

**Comment by simonpoole** [comment]
A I mentioned in my last PR we currently don't have agreement on how to categorize IR/NIR/CIR imagery see also https://github.com/osmlab/editor-layer-index/issues/325 @don-vip @Klumbumbus how are you categorizing these?

**Comment by don-vip** [comment]
What are "IR/NIR/CIR" layers?

**Comment by simonpoole** [comment]
Infrared and Near Infrared

**Comment by imagico** [comment]
As mentioned in https://github.com/osmlab/editor-layer-index/pull/324#issuecomment-300546509 all infrared *aerial* imagery in the index is currently false color images with a green-red-nir band combination, most infrared *satellite* imagery is a red-nir-swir combination.

Other possibilities for images based on infrared data could be monochrome NIR images or visualizations of computed values like NDVI.

The main use of storing such information in differentiated form would be to allow editors to show users specific instructions how to interpret the imagery.

**Comment by Klumbumbus** [comment]
> @Klumbumbus how are you categorizing these?

As photo or historicphoto. If a photo is true or false color is not covered by the current category values.

**Comment by Hedaja** [comment]
Another good category would be DSM/DEM/DTM for (lidar-scannend) surface models. 
These are extremely helpful when mapping in forested areas.

**Comment by simonpoole** [comment]
"elevation' already includes those (l'm not sure why this issue hasn't long been closed).
